src/Components/ArticlePage.js

```diff
@@ -0,0 +1,56 @@
+import "./ArticlePage.css";
+import { useParams, useLocation } from "react-router-dom";
+import { getArticleById } from "../Api/Api";
+import React, { useEffect, useState } from "react";
+
+const ArticlePage = (props) => {
+  const { id } = useParams();
+  const { state } = useLocation();
```

Comment:
I would strongly advise against this - it leads you to be unaware of what article is in your state and how the user arrived at this path should ideally not make a difference.
Where navigating, consider using a regular Link component and fetching the article in here